cartographer/io/mapping_state_deserializer_test.cc

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+
+#include "cartographer/io/internal/in_memory_proto_stream.h"
+#include "cartographer/io/internal/testing/serialized_mapping_state_text_proto.h"
+#include "cartographer/io/mapping_state_deserializer.h"
+#include "glog/logging.h"
+#include "gmock/gmock.h"
+#include "google/protobuf/text_format.h"
+#include "google/protobuf/util/message_differencer.h"
+#include "gtest/gtest.h"
+
+namespace cartographer {
+namespace io {
+namespace {
+
+using common::make_unique;
+using google::protobuf::Message;
+using mapping::proto::AllTrajectoryBuilderOptions;
+using mapping::proto::PoseGraph;
+using mapping::proto::SerializationHeader;
+using mapping::proto::SerializedData;
+using ::testing::Eq;
+using ::testing::Not;
+
+void ProtoFromStringOrDie(const std::string& proto_string,
+                          google::protobuf::Message* msg) {
+  CHECK(google::protobuf::TextFormat::ParseFromString(proto_string, msg));
+}
+
+template <typename T>
+T ProtoFromStringOrDie(const std::string& proto_string) {
+  T msg;
+  CHECK(google::protobuf::TextFormat::ParseFromString(proto_string, &msg));
+  return msg;
+}
+
+template <typename T>
+std::unique_ptr<T> CreateProtoUPtrFromString(const std::string& proto_string) {
```

Comment:
Rename to CreateProtoUniquePtrFromString

Reply:
renamed to `ProtoUPtrFromStringOrDie`